### PR TITLE
Fix: handle nil expression in `evaluate_as_string`

### PR DESCRIPTION
**Pull request by maryamaljanabi**
`evaluate_as_string/3` method fails when `expression` is `nil`. There are cases when `expression` is sent as `nil` since this method is called from `flow_runner` and various methods on `engage` backend where this case happens.

This PR handles the `nil` case in this method and adds the required test.

### CODE REVIEW

**Comment by maryamaljanabi** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by maryamaljanabi** [comment]
recheck

File: lib/expression.ex [hunk 1]
```diff
@@ -120,11 +120,15 @@ defmodule Expression do
   end
 
   def evaluate_as_string!(expression, context \\ %{}, mod \\ Expression.Callbacks) do
```
Comment (smn) [inline lib/expression.ex]:
```suggestion
  def evaluate_as_string!(expression, context \\ %{}, mod \\ Expression.Callbacks)
  def evaluate_as_string!(nil, _context, mod), do: ""
  def evaluate_as_string!(expression, context, mod) do
```

Thanks @maryamaljanabi the change is correct but could we use pattern matching here?

**Review by smn — APPROVED** [review]
Approving because it's correct :)

**Comment by smn** [comment]
could you also attend to the formatting change required @maryamaljanabi  ?

**Comment by smn** [comment]
@maryamaljanabi don't worry about the CLA assistant, I can merge it on your behalf. That thing is a bit buggy at times.

File: lib/expression.ex (hunk 1 above)
Comment (maryamaljanabi) [reply lib/expression.ex]: Thanks a lot, I'll make an update

**Comment by maryamaljanabi** [comment]
@smn I've made the update and tests are passing. I have also tested that these changes will solve the `nil` bug with Santiago's suggestion on Slack. The CLA is still failing though, is there something specific I should commit for it to pass?

**Comment by smn** [comment]
no I think this is great, the fact that the CLA is failing is strange. I made a change and now a second CLA thing is passing -- hopefully that'll solve it for the next PR. Please go ahead and merge!
